src/utils/components/LazyActionMenu/overrides.ts

```diff
@@ -0,0 +1,138 @@
+ * Note that the loading state will be displayed only during first call to checkAccess().
+ * Subsequent calls will update the `isAllowed` flag but not the `loading` state.
+ * The provided checkAccess() implementation should be memoized.
+ * 
+ * @param resourceAttributes resource attributes for access review
+ * @param impersonate impersonation details
+ * @param checkAccessDelegate defaults to built-in checkAccess() implementation
+ * @returns Array with `isAllowed` and `loading` values.
+ 
+ */
+export const useCheckAccess = (
+  resourceAttributes: AccessReviewResourceAttributes,
+  impersonate?: ImpersonateKind,
+  checkAccessDelegate: CheckAccess = checkAccess,
+): [boolean, boolean] => {
+  const [loading, setLoading] = useState(true);
+  const [isAllowed, setAllowed] = useState(false);
+
+  useEffect(() => {
+    checkAccessDelegate(resourceAttributes, impersonate)
+      .then((result: SelfSubjectAccessReviewKind) => {
+        setAllowed(result.status.allowed);
+        setLoading(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        kubevirtConsole.warn('SelfSubjectAccessReview failed', e);
+        // Default to enabling the action if the access review fails so that we
+        // don't incorrectly block users from actions they can perform. The server
+        // still enforces access control.
+        setAllowed(true);
+        setLoading(false);
+      });
+  }, [resourceAttributes, impersonate, checkAccessDelegate]);
+  return [isAllowed, loading];
+};
+
+export const createLocalMenuOptions = (actions: ActionDropdownItemType[]): MenuOption[] =>
+  actions?.map((action) =>
+    action.options?.length > 0
+      ? {
+          ...action,
+          children: createLocalMenuOptions(action.options),
+          // action groups should have no attached direct actions
+          cta: undefined,
+          // default to submenus
+          submenu: true,
+        }
+      : action,
+  ) ?? [];
+
+/**
+ * Merge duplicated options that exist on the same level.
+ * For actions - take the first defined.
+ * For groups - take the props from the first, merge the children.
+ * @param options list of options (with duplicates on the same level)
+ * @returns merged, de-duplicated list (on the same level)
+ */
+export const mergeOptions = (options: MenuOption[]): MenuOption[] =>
+  // use array based dictionary to guaratnee the order
+  // first encountered is first in the results
+  options
+    .reduce((acc, opt) => {
+      const existingIndex = acc.findIndex(({ id }) => id === opt.id);
+      if (existingIndex === -1) {
+        acc.push({ duplicates: [opt], id: opt.id });
+      } else {
+        acc[existingIndex].duplicates.push(opt);
+      }
+      return acc;
+    }, [] as { duplicates: MenuOption[]; id: string }[])
+    .map(
+      ({ duplicates }): MenuOption => ({
+        // grouping is stable, take the props from the first defined option
+        ...duplicates[0],
```

Comment:
.?

Reply:
To create a new group we need at least one item - the code looks safe.